route.go

```diff
@@ -6,9 +6,27 @@ import (
 	"sync/atomic"
 )
 
-type RouteConfig struct {
-	Matcher Matcher
-	Dests   []*Destination
+type RouteConfig interface {
+	Matcher() *Matcher
+	Dests() []*Destination
+}
+
+type baseRouteConfig struct {
+	matcher Matcher
+	dests   []*Destination
+}
+
+func (c baseRouteConfig) Matcher() *Matcher {
+	return &c.matcher
+}
+
+func (c baseRouteConfig) Dests() []*Destination {
+	return c.dests
+}
+
+type consistentHashingRouteConfig struct {
+	RouteConfig
```

Comment:
i didn't know go lets you embed an interface type? i thought you'd embed baseRouteConfig here. how does this work?